File: cardbattlemodule/src/main/java/com/syahputrareno975/cardbattlemodule/interfaces/ShopStreamEvent.java
package com.syahputrareno975.cardbattlemodule.interfaces;

import com.syahputrareno975.cardbattlemodule.model.card.CardModel;
import com.syahputrareno975.cardbattlemodule.model.playerWithCard.PlayerWithCardsModel;

import java.util.ArrayList;

public interface ShopStreamEvent {
    void onConnected(ShopStreamEventController c);
    void onGetPlayerData(PlayerWithCardsModel p);
    void onAllCardInShop(ArrayList<CardModel> c);
    void onCardBought( Boolean success);
    void onCardSold( Boolean success);
    void onCardUpgraded(Boolean success);
    void onAddCardSlot(Boolean success);
    void onShopCountDown(int i);
    void onShopRefreshed();
    void onDisconnected();
    void onError(String e);
    void onException(String e,int flag,ShopStreamEventController c);
}

File: app/src/main/java/com/syahputrareno975/simplecardbattle/interfaces/ShopStreamEvent.kt
package com.syahputrareno975.simplecardbattle.interfaces

import com.syahputrareno975.simplecardbattle.model.card.CardModel
import com.syahputrareno975.simplecardbattle.model.playerWithCard.PlayerWithCardsModel

interface ShopStreamEvent {
    fun onConnected(c : ShopStreamEventController)
    fun onGetPlayerData(p : PlayerWithCardsModel)
    fun onAllCardInShop(c : ArrayList<CardModel>)
    fun onCardBought(success : Boolean)
    fun onCardSold(success : Boolean)
    fun onCardUpgraded(success : Boolean)
    fun onAddCardSlot(success : Boolean)
    fun onShopCountDown(i : Int)
    fun onShopRefreshed()
    fun onDisconnected()
    fun onError(e : String)
    fun onException(e : String,flag : Int,c : ShopStreamEventController)
}